### DISCUSSION TITLE
Potential Sync Fix Coming in 1.6 ?

### DISCUSSION THREAD

**Discussion by insomnolence** [Chia-Network/chia-blockchain#5631] [Support]
I, like so many, have had issues staying in sync. I'm lucky enough to have gotten into sync, but after 40 mins, it all falls apart. I'd shutdown the nodes and restart . . . and again 40 mins later, out of sync.

Recently, I noticed that there was a fix put in #5590 "Speedup normal sync and wallet sync". So, hopefully, the next code drop ( 1.6 ) should help fix a lot of the issues in general with syncing ( and a whole slew of other fixes as well ).

Hopefully it drops soon to help folks out.

BUT 

---- The following is a AT YOUR OWN RISK and I TAKE NO RESPONSIBILITY if you have to re-sync from scratch ----

Following the discussion in that thread, I took a gamble and took my nodes offline, brought up the databases with an SQLite browser and ran analyze on them. Then saved them back to disk. ( Of course I backed them up first !! ) I haven't had a sync issue since.

Given the info in the #5590 thread, it seems like there is a potential for a long/missed database query and the analyze command will look at the tables and optimize the lookups. Since my current database is about 3.2 Gigs in size, slow/missed lookups would compound over time ( maybe, say . . . in about 40ish minutes ? ) and just have the node fall behind and out of sync. This could be an issue for folks that have never sync'd. The longer the nodes are online, the further behind they get ( especially as the database grows ).

Your best ( and safest bet ) is to wait until the 1.6 code drop as it looks like they've modified the database lookup. But, if you want to gamble and risk having to re-sync, you could try and run the sqlite analyze command on the databases and see if that helps you. I can take no credit for the idea, as it was in the discussion thread. I was just foolish enough to try it :)

Oh, did I mention ?  Try it at your own risk !! I can't imagine having to re-sync from scratch.

**Comment by BrandtH22** [comment]
@insomnolence , if you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
